Source/WebCore/rendering/RenderObject.cpp

```diff
@@ -2426,30 +2438,19 @@ auto RenderObject::collectSelectionGeometriesInternal(const SimpleRange& range)
         }
     }
 
-    auto numberOfGeometries = geometries.size();
-
-    // If the selection ends in a BR, then add the line break bit to the last rect we have.
-    // This will cause its selection rect to extend to the end of the line.
-    if (numberOfGeometries) {
-        // Only set the line break bit if the end of the range actually
-        // extends all the way to include the <br>. VisiblePosition helps to
-        // figure this out.
-        if (is<HTMLBRElement>(VisiblePosition(makeContainerOffsetPosition(range.end)).deepEquivalent().firstNode()))
-            geometries.last().setIsLineBreak(true);
-    }
-
-    int lineTop = std::numeric_limits<int>::max();
-    int lineBottom = std::numeric_limits<int>::min();
+    // Do line 0 first, and then check the boundaries of each line after.
+    int lineTop = geometries[0].logicalTop();
+    int lineBottom = lineTop + geometries[0].logicalHeight();
     int lastLineTop = lineTop;
     int lastLineBottom = lineBottom;
-    int lineNumber = 0;
+    int lineNumber = geometries[0].isRubyText() ? 0 : 1;
```

Comment:
The question is, if line 0 is ruby text, should we count it? The previous code says yes, but I do not see any reason or documentation or any history that would explain why.

Reply:
AKA isRubyText() means we do not count lineNumber 0 and instead count the line below it as 0, but originally line 0 always was counted regardless because the check was only done if i was not 0, as in not the first case.